### ISSUE TITLE
Some reactions that was added by gap fill have good blast and high coverage 

### ISSUE THREAD

**Issue by Zhelunnn** [jotech/gapseq#115]
> for bug reports and errors please report the output of: ./gapseq test
Hi,

Thank you for developing gapseq. I just started using it. I got my gapfill results and was investigating which reactions were added in my final model. I found some of the filled reactions have good-blast and very high coverage score. As I known, they should be listed in gapfind file '-all-Reactions.tbl'. I read your paper and didn't find things about this. I am not sure that if I should classify them into gapfill result, which would affect my recognition on my model. Could you have any comment on this? 

Thank you and best,
Zhelun

**Comment by Waschina** [comment]
Hi Zhelun,

there could be different causes:

1. Some reactions are flagged by gapseq to require a higher sequence identity in order to be predicted as present already before gapfilling. These reactions are listed in the file https://github.com/jotech/gapseq/blob/master/dat/exception.tbl
2. If a reaction in catalysed by an enzyme complex, enough subunits need to be found in order to call the reaction present.

Thus, in the case you described I would classify all reactions that are added to the model in the gapfilling step also as gapfilled reactions. You might also find the 'reaction tracing' described [here](https://gapseq.readthedocs.io/en/latest/tutorials/traceability.html) helpful – gapseq tracks for each reaction why it was added to the model.

Hope this helps.

Best wishes
Silvio

**Comment by Zhelunnn** [comment]
Hi Silvio
Thank you for this. But I think my case is not in the two causes above. 
1. There are many reactions annotated as good blast (more than 40 in one model). Almost all of them can not be found in 'exception.tbl'.
2. I have to say most of their 'complex' annotations are  'NA'. I can see some of them were annotated as subunit 1 or subunit undefined in this column. The 'gs.origin' of all these reactions are filled by '1'. 
3. I can also find most of these nucleotide sequence were annotated as other reactions which looks similar with gapfill one in ModelSeed database.
It seems this enzyme can catalyze many reactions, but some of them were not added in the draft model. Is that what I think?

Thank you,
Zhelun

**Comment by Waschina** [comment]
Hi Zhelun,

your point 3 could be the explanation. Often an ORF in the genome has 'good hits' to more than one reaction. In some of those cases, gapseq associates these ORFs only to the reaction with the highest alignment bitscore. We identified these cases during the benchmarking of gapseq and implemented the 'best hit only' handling for some reactions, for which we found no evidence in the literature, that these reactions can be catalyzed by the same promiscuous enzyme. I have to admit that this reaction-hit conflict handling is currently hard-coded in gapseq. I will improve this in the way that there is a table in the `dat/` directory that lists these cases.
